odfuzz/config.py

```diff
@@ -1,8 +1,24 @@
 """This module contains classes for fetching and parsing basic configurations."""
 
+import importlib
```

Comment:
why is this change needed?

Reply:
This module `importlib` is used for importing modules which are out of the scope during the runtime. I am using it to get a function object from the configuration specified by a user. The configuration looks like this: https://github.com/SAP/odfuzz/pull/47/files#diff-31a9af601fb02ccdbe2b445fc0628e6eR53-R54

Reply:
But is it really necessary? I mean, " importing modules which are out of the scope during the runtime." looks like a problem on its own and the importlib is just a bandage slapped on it. 

- This will most probably break any static analysis relevant to inner dependencies (e.g. pydeps), which we need to emply anyway to better separate modules, instead of cyclic dependencies.
- If adding configuration in current state of codebase is such a problem, I would prefer to just directly fix the encoded without any configuration and open a new issue for such. 
- in my opinion, it would be much simpler and direct to just use true constant for DEFAULT_STRING_ENCODER/DECODER, instead of passing function back and forth and solve the import and probably gevent related problems.  

I would like to move in a way that the runtime stuff is one block, isolated away from fuzzer, which should be perhaps even a true isolated python library, for generating URL from odata V2/V4 only - essencially entities, generators, mutators.  It will be more testable and not inertwined with stuff like networking, greenlets or database. 

If you agree with this vision that could be better, then is obviously a bad call to add something like Config.fuzzer.decode_string(search_option  and call it in fuzzer.py, while in effect it is about generators and mutators and the string encoder change propagated there as well.

Reply:
seems that we get anywehere in this  long question than "it just stays there" :)

Reply:
The idea behind this approach was to load any encoder/decoder which will properly serve for encoding and decoding. What if a user wants to define his own decoder? The module config.py should be used to initialize the configuration for ODfuzz. As a part of that, it just transforms string value defined by a user to a proper python object that is going to be used during the runtime.

Reply:
"What if user wants to define his own X"... this is argument IMHO similar to premature optimization; premature scope creep when there is at the moment not even notable user base for odfuzz as currently is. If we approach any fix with this argument, we wold sunk in configuration to the state that we provide just a "inner platform/own language" for plugins etc. 

My feeling from this is that we started with providing option to user to modify some inner  parts too early and also on wrong end at the moment.  It is a CLI tool after all and it should "just work", since such is the way it is used. It should therefore work well (existing bugs and not covered odata data types or odata protocol versions) before it is necessary to think like "work well in any way user wishes".

We could use similar approach to other areas, that's for sure.. for example the ability to provide own pseudorandom algorithm or some sequence. This would be great in the state of codebase described two three comments above.. library, perhaps even on pip, just for fuzzing URLs based on metadata and other inputs.. and separate project for current CLI runtime with requests and mongo using such library code. 

I think it is a great prototype for how things could be configured, but at the moment, I am more leaning towards no configuration at all, it is a bugfix PR after all. Or if really seems to be necessary to have ability to turn it off for whatever reason I cannot image, just simple boolean value should be enough.

I also think that may be better to apply the the string replace in one place only.. where the url parts are "joining together" in fuzzer.py. Or is there a reason to call in in mutator, generator and fuzzer also? IMHO there should not be problem that generator and mutator creates strings with UTF-8 characters that are not OK for URL parts... encode the url in url-related stuff and just once, right? Or just in mutator and generator and create contract that those modules provide encoded URL parts.

I am looking at this PR and #45 in comparison - quite similar problem, different scope of solving the problem.